clientv3/lease.go

```diff
@@ -216,6 +217,13 @@ func (l *lessor) KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAl
 	ch := make(chan *LeaseKeepAliveResponse, leaseResponseChSize)
 
 	l.mu.Lock()
+	// ensure that recvKeepAliveLoop is still running
+	select {
+	case <-l.donec:
+		l.mu.Unlock()
+		return nil, errors.New("keep alive loop exited")
```

Comment:
make this a constant `var ErrLeaseHalted = errors.New("etcdclient: leases halted")` at the top of the file?

Also `close(ch)` and return that instead of `nil` as well?